app/src/main/java/org/sil/gatherwords/EntryFragment.java

```diff
@@ -61,6 +63,25 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         return entryPage;
     }
 
+    /**
+     * Will run after the view is created and after the picture is taken.
+     */
+    @Override
+    public void onResume() {
+        super.onResume();
+        new UpdateAfterPicture(getView()).execute(m_wordID);
+    }
+
+    /**
+     * Gets the current wordId
+     *
+     * Sometimes m_wordID is not yet set for some reason
+     * @return The wordID of the current Word
+     */
+    public long getWordID() {
+        return getArguments().getLong(ARG_WORD_ID);
```

Comment:
Why is this necessary?  In what situation is `m_wordID` not set?

Reply:
When I first made this, after figured out how to get the correct fragment, I kept getting a `0` when I should not have. I'll try again though

Reply:
Not certain why this occurs, but it seems to be only upon returning from taking a photo.  Probably update the comment to specify this, and call it good enough.